### ISSUE TITLE
UnsupportedOperationException when mining EXU2 Machine

### ISSUE THREAD

**Issue by OrdinatorStouff** [Landmaster/PlusTiC#149]
PlusTiC: `plustic-6.3.2.1.jar`
Forge: `forge-1.12.2-14.23.1.2587-universal.jar`
ExU2: `extrautils2-1.12-1.7.2.jar`
Pack: ATM3 5.7b, Dedi Server

Possibly related to https://github.com/Landmaster/PlusTiC/issues/15 but this is happening to us still/now and reproduceable on our server.

Player mines an ExU2 "Furnace Generator" with speed upgrades in it (not sure if relevant) with global traveler tool and server crashes. On server reload, the furnace generator is gone. Placed a new generator down with upgrades to test if it could be reproduced and crashed again the same manner.

Coordinates of the global traveler output is an Actually Additions Storage Crate.

[Crash Log](https://gist.github.com/OrdinatorStouff/14fd90a36f31bed1bb56dd92165a60d8#file-crash-2018-01-31_22-00-24-server-txt)
[tconstruct:pickaxe NBT](https://gist.github.com/OrdinatorStouff/14fd90a36f31bed1bb56dd92165a60d8#file-tconstruct-pickaxe-nbt-json)
[playerdata.zip](https://github.com/Landmaster/PlusTiC/files/1683571/playerdata.zip)

**Comment by Landmaster** [comment]
It looks like ExU2 is overriding the officially deprecated method `getDrops(IBlockAccess, BlockPos, IBlockState, int)` instead of the correct method `getDrops(NonNullList, IBlockAccess, BlockPos, IBlockState, int)`; it should be overriding the latter, not the former. (The latter function was created to avoid issues such as #15.)

The reason why I introduced the coremod fix for 1.10.2 and #15 is that prior to 1.11.2, there was no way to enforce that modders correctly return a mutable list from the now-deprecated function, meaning that I had to resort to that hack. Things have changed, however, and now one should reasonably expect that other modders correctly override the new function, which is why I removed the coremod in 1.12.

Normally, I would tell you to report this to ExU2. However, ExU2 is not only closed-source, but also notorious for flaunting the conventions of maintainable modding practices (e.g. using the aforementioned deprecated method). As a rule, I usually do not support closed-source mods (including ExU2) due to how d*mn frustrating it is to debug mod interactions with them (since I cannot view their source). To make matters worse, ExU2's issue tracker is practically stale, and you can blame @RWTema for it.

**TL;DR: Just don't mine ExU2 blocks with Alumite; I don't support ExU2.**

**Comment by OrdinatorStouff** [comment]
*There's a reason why I came to your issue tracker and not Tema's ;)*

I appreciate the explanation and thorough feedback. I find your response more than reasonable and at the very least by means of this report you are now informed of the current interaction "incompatability" should anyone else mention it or report it to you in the near future.

Thanks again, Cheers!

**Comment by Landmaster** [comment]
@OrdinatorStouff Most likely, I will put a temporary workaround in the mod in the next version of PlusTiC to simply log a warning when breaking such blocks instead of crashing the game, just in case someone hostile wants to deliberately crash the server.

**Comment by Landmaster** [comment]
@rwtema released source code for Extra Utilities a few months ago, and it appears that Extra Utilities has fixed this issue on its end—at long last.
